Libraries/LibTLS/Socket.cpp

```diff
@@ -184,6 +132,57 @@ bool TLSv12::common_connect(const struct sockaddr* saddr, socklen_t length)
     return true;
 }
 
+void TLSv12::write_into_socket()
+{
+#ifdef TLS_DEBUG
+    dbg() << "Flushing cached records: " << m_context.tls_buffer.size() << " established? " << is_established();
+#endif
+    m_has_scheduled_write_flush = false;
+    if (!check_connection_state(false))
+        return;
+    flush();
+
+    if (!is_established()) {
```

Comment:
Unfortunately this little bit is needed to start the AppData transfer, I'll need to look a bit into what can be done about it.